hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -222,53 +231,203 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
 
     String toDatabase = to.namespace().level(0);
-    String fromDatabase = from.namespace().level(0);
-    String fromName = from.name();
+    String fromDatabase = fromIdentifierEntity.namespace().level(0);
+    String fromName = fromIdentifierEntity.name();
 
     try {
-      Table table = clients.run(client -> client.getTable(fromDatabase, fromName));
-      HiveOperationsBase.validateTableIsIceberg(table, fullTableName(name, from));
+      Table fromEntity = clients.run(client -> client.getTable(fromDatabase, fromName));
+      if (entityType.equalsIgnoreCase("Table")) {
+        HiveOperationsBase.validateTableIsIceberg(
+            fromEntity, CatalogUtil.fullTableName(name, fromIdentifierEntity));
+      } else {
+        HiveOperationsBase.validateTableIsIcebergView(
+            fromEntity, CatalogUtil.fullTableName(name, fromIdentifierEntity));
+      }
+
+      validateToTableForRename(fromIdentifierEntity, to);
 
-      table.setDbName(toDatabase);
-      table.setTableName(to.name());
+      fromEntity.setDbName(toDatabase);
+      fromEntity.setTableName(to.name());
 
       clients.run(
           client -> {
-            MetastoreUtil.alterTable(client, fromDatabase, fromName, table);
+            MetastoreUtil.alterTable(client, fromDatabase, fromName, fromEntity);
             return null;
           });
 
-      LOG.info("Renamed table from {}, to {}", from, to);
+      LOG.info("Renamed {} from {}, to {}", entityType, fromIdentifierEntity, to);
 
-    } catch (NoSuchObjectException e) {
-      throw new NoSuchTableException("Table does not exist: %s", from);
+    } catch (NoSuchObjectException | NoSuchViewException e) {
+      if (entityType.equalsIgnoreCase("Table")) {
+        throw new NoSuchTableException("Table does not exist: %s", fromIdentifierEntity);
+      } else {
+        throw new NoSuchViewException(
+            "Cannot rename %s to %s. View does not exist", fromIdentifierEntity, to);
+      }
 
     } catch (InvalidOperationException e) {
       if (e.getMessage() != null
           && e.getMessage().contains(String.format("new table %s already exists", to))) {
         throw new org.apache.iceberg.exceptions.AlreadyExistsException(
-            "Table already exists: %s", to);
+            "%s already exists: %s", entityType, to);
       } else {
-        throw new RuntimeException("Failed to rename " + from + " to " + to, e);
+        throw new RuntimeException("Failed to rename " + fromIdentifierEntity + " to " + to, e);
       }
 
     } catch (TException e) {
-      throw new RuntimeException("Failed to rename " + from + " to " + to, e);
+      throw new RuntimeException("Failed to rename " + fromIdentifierEntity + " to " + to, e);
 
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
       throw new RuntimeException("Interrupted in call to rename", e);
     }
   }
 
+  private void validateToTableForRename(TableIdentifier from, TableIdentifier to)
```

Comment:
I have moved this to now renameContent as [here](https://github.com/apache/iceberg/pull/9852/files#diff-3fdd21d335b598f5354de287be6a8830d78e381c89274f0d6bf85c4c9966b8f0R386-R393) . 
This was needed to throw error with explicit contentType in case of `renameView`.